contrib/pg_tde/src/access/pg_tde_xlog_smgr.c

```diff
@@ -266,16 +266,33 @@ TDEXLogSmgrInitWrite(bool encrypt_xlog)
 		pfree(key);
 }
 
+/*
+ * Used by pg_tde_restore_encrypt to simulate being constantly in recovery
+ * since the command does not have access to any information about if we are in
+ * recovery or not.
+ *
+ * Creates a dummy key which points at the very end of the WAL stream.
+ */
 void
-TDEXLogSmgrInitWriteReuseKey()
+TDEXLogSmgrInitWriteOldKeys()
 {
-	WalEncryptionKey *key = pg_tde_read_last_wal_key();
+	WALKeyCacheRec *keys;
+	WalEncryptionKey dummy = {
+		.type = WAL_KEY_TYPE_UNENCRYPTED,
+		.wal_start = {.tli = -1,.lsn = -1}
```

Comment:
I'd rather use something like `(~(XLogRecPtr)0)` and `(~(TimeLineID)0)` (or better named constants like [here](https://github.com/percona/postgres/blob/80e0bb0b56a2d64cdced770ded2da19003eef5ca/contrib/pg_tde/src/access/pg_tde_xlog_keys.c#L28-L29)) just for the reader's clarity sake. It's not obvious on the first glance that .tli and .lsn are unsigned types.